server/eclipse-project/src/main/java/us/freeandfair/corla/model/CountyContestComparisonAudit.java

```diff
@@ -486,11 +528,35 @@ private BigDecimal computeOptimisticBallotsToAudit(final int the_two_under,
   }
   
   /**
-   * Signals that a new ballot has been audited. This ensures that estimates
-   * are recalculated correctly.
+   * Signals that a sample has been audited. This ensures that estimates
+   * are recalculated correctly and states are updated.
+   * 
+   * @param the_count The count of samples that have been audited simultaneously
+   * (for duplicates).
    */
-  public void signalBallotAudited() {
+  public void signalSampleAudited(final int the_count) {
     my_estimated_recalculate_needed = true;
+    my_audited_sample_count = my_audited_sample_count + the_count;
+    
+    if (my_audit_status != AuditStatus.ENDED) {
+      my_audit_status = AuditStatus.IN_PROGRESS;
+    }
+  }
+  
+  /**
+   * Signals that a sample has been unaudited. This ensures that estimates
+   * are recalculated correctly and states are updated.
+   * 
+   * @param the_count The count of samples that have been unaudited simultaneously
+   * (for duplicates).
+   */
+  public void signalSampleUnaudited(final int the_count) {
+    my_estimated_recalculate_needed = true;
+    my_audited_sample_count = my_audited_sample_count - the_count;
+    
+    if (my_audit_status != AuditStatus.ENDED) {
+      my_audit_status = AuditStatus.IN_PROGRESS;
```

Comment:
With this, starting with no audited samples, if we did:
```java
signalSampleAudited(1);
signalSampleUnaudited(1);
```
That leave us in a state where the audit status is `IN_PROGRESS` but `my_audited_sample_count` is 0. Should we prevent that? Does it matter, in context?

Reply:
I suspect it doesn't matter in context... the audit really is "in progress", because you've started it and haven't stopped it, and in actual usage `signalSampleAudited` would be called again right after `signalSampleUnaudited`.

Reply:
Yeah, that part makes sense. I guess we'll just have to be careful not to incorrectly assume that "in progress" implies "at least one sample audited".

Reply:
Given that the sample count is actually part of the record, I suspect that won't be a problem. This is likely to be only used in reporting anyway.